QMutex recursion mode is deprecated

**Describe the bug**
Newer QT version seems to have deprecated QMutex recursion mode.

**Steps to Reproduce**
Steps to reproduce the behavior:
1. Use the live 5.15 qtcore git branch
2. Try to compile mumble
3. Compilation fails with the following error:

```
/var/tmp/portage/media-sound/mumble-9999/work/mumble-9999/src/mumble/Plugin.cpp:18:45: error: ‘QMutex::QMutex(QMutex::RecursionMode)’ is deprecated: Use QRecursiveMutex instead of a recursive QMutex [-Werror=deprecated-declarations]
   18 | QMutex Plugin::s_idLock(QMutex::NonRecursive);
      |                                             ^
In file included from /usr/include/qt5/QtCore/QMutex:1,
                 from /var/tmp/portage/media-sound/mumble-9999/work/mumble-9999/src/mumble/Plugin.h:15,
                 from /var/tmp/portage/media-sound/mumble-9999/work/mumble-9999/src/mumble/Plugin.cpp:6:
/usr/include/qt5/QtCore/qmutex.h:140:14: note: declared here
  140 |     explicit QMutex(RecursionMode mode);
      |              ^~~~~~
cc1plus: all warnings being treated as errors
```

**Desktop (please complete the following information):**
 - OS: Gentoo Linux
 - Version: Mumble git master

--- a/src/mumble/Plugin.cpp
+++ b/src/mumble/Plugin.cpp
@@ -15,7 +15,7 @@
 
 // initialize the static ID counter
 plugin_id_t Plugin::s_nextID = 1;
-QMutex Plugin::s_idLock(QMutex::NonRecursive);
+QMutex Plugin::s_idLock;
 
 void assertPluginLoaded(const Plugin *plugin) {
 	// don't throw and exception in release build

--- a/src/mumble/ServerHandler.cpp
+++ b/src/mumble/ServerHandler.cpp
@@ -59,7 +59,7 @@
 
 // Init ServerHandler::nextConnectionID
 int ServerHandler::nextConnectionID = -1;
-QMutex ServerHandler::nextConnectionIDMutex(QMutex::Recursive);
+QMutex ServerHandler::nextConnectionIDMutex;
 
 ServerHandlerMessageEvent::ServerHandlerMessageEvent(const QByteArray &msg, unsigned int mtype, bool flush)
 	: QEvent(static_cast< QEvent::Type >(SERVERSEND_EVENT)) {

